src/lib/ext-activitylog.js

```diff
@@ -3,26 +3,182 @@ import { save } from './save-load.js';
   }
 }
 
 class View {
   constructor() {
-    this.logTable = document.querySelector('log-view');
-    this.saveLogBtn = document.getElementById('saveLogBtn');
+    this.logView = document.querySelector('log-view');
+    this.saveLogBtn = document.querySelector('#saveLogBtn');
     this.notice = document.querySelector('.notice');
+    this.checkboxTemplate = document.querySelector('#filterCheckbox').content;
 
-    this.saveLogBtn.addEventListener('click', function () {
-      this.dispatchEvent(new CustomEvent('savelog'));
-    });
+    this.extCheckboxList = document.querySelector('.checkbox-list.extension');
+    this.extFilterBtn = document.querySelector('.toggle-btn.extension');
+
+    this.viewTypeCheckboxList = document.querySelector(
+      '.checkbox-list.viewtype'
+    );
+    this.viewTypeFilterBtn = document.querySelector('.toggle-btn.viewtype');
+
+    this.extFilterBtn.addEventListener('click', this);
+    this.extCheckboxList.addEventListener('change', this);
+    this.viewTypeFilterBtn.addEventListener('click', this);
+    this.viewTypeCheckboxList.addEventListener('change', this);
+    this.saveLogBtn.addEventListener('click', this);
+  }
+
+  handleEvent(event) {
+    if (event.type === 'click') {
+      switch (event.currentTarget) {
+        case this.saveLogBtn:
+          this.saveLogBtn.dispatchEvent(new CustomEvent('savelog'));
+          break;
+        case this.extFilterBtn:
+          this.toggleFilterListDisplay(this.extFilterBtn, this.extCheckboxList);
+          break;
+        case this.viewTypeFilterBtn:
+          this.toggleFilterListDisplay(
+            this.viewTypeFilterBtn,
+            this.viewTypeCheckboxList
+          );
+          break;
+        default:
+          throw new Error(`wrong event target - ${event.target.tagName}`);
+      }
+    } else if (event.type === 'change') {
+      this.handleChangeOnCheckbox(event);
+    } else {
+      throw new Error(`wrong event type - ${event.type}`);
+    }
+  }
+
+  toggleFilterListDisplay(toggleBtn, checkboxList) {
+    checkboxList.hidden = !checkboxList.hidden;
+    this.toggleFilterBtnIcon(toggleBtn, checkboxList.hidden);
+  }
+
+  toggleFilterBtnIcon(element, isListHidden) {
+    isListHidden
+      ? element
+          .querySelector('.arrow-down')
+          .classList.replace('arrow-down', 'arrow-right')
+      : element
+          .querySelector('.arrow-right')
+          .classList.replace('arrow-right', 'arrow-down');
+  }
+
+  handleChangeOnCheckbox(event) {
+    let target = null;
+    switch (event.currentTarget) {
+      case this.extCheckboxList:
+        target = 'id';
+        break;
+      case this.viewTypeCheckboxList:
+        target = 'viewType';
+        break;
+      default:
+        break;
+    }
+
+    if (target) {
+      const filterObj = {
+        filterParam: { value: event.target.closest('input').value, target },
+        checked: event.target.closest('input').checked,
```

Comment:
> This `checked` property is very specific to the UI element (checkbox) here. The purpose of firing the event is to abstract the actual UI implementation.

I have renamed the `checked` property with `isFilterRemoved`. Is this okay?

> `filterParam`, `value` and `target` are bad names, because it is not at all obvious what those are supposed to be used for.

These names have been updated. :-)

> As an example, if you trigger the `filter:remove` event with detail `{logKey: "viewType", valueEquals: "popup"}`, then consumers can listen for the event (instead of having to translate the specifics of the UI element to a logical representation, as you're doing right now). I used `filter:` as a prefix, so that there is a namespace where relateds events can be grouped, e.g. `filter:add`, `filter:remove`, ...

I am not sure if I have fully understood this above part, did you want me to make different events?